src/work.rs

```diff
@@ -529,7 +533,7 @@ impl<'a> Work<'a> {
                     }
                 };
                 if mtime == MTime::Missing {
-                    if workaround_missing_phony_deps {
+                    if workaround_missing_phony_deps || self.dry_run {
```

Comment:
I think in the dry-run case here we should return Ok(true), see the function comments about what that value means.  (In particular we want to avoid hashing when there's a file missing, because there's a panic when that happens.)

Reply:
Done